### PR TITLE
Manage Token Map for DGA

### LINKED ISSUE #75: [BUG] FDGA: Crashes when `token_map_path` file has wrong format
**Describe the bug**
When the file contained in the `"token_map_path"` field of the config is not properly formatted the filter crashes without logging.

**Platform (please complete the following information):**
 - OS (version): FreeBSD (12.0)
 - Darwin version: 1.0

**To Reproduce**
1. Generate a random file
2. Set the field `"token_map_path"` to ths file
3. Start fdga

**Expected behavior**
Log a CRITICAL error with the proper message and quit gracefully.

**Additional context**
Hint: Look around line 140-143 of fdga/Generator.cpp

### PR DESCRIPTION
**Pull request by NS4nti**
# :sparkles: Pull Request Template
:bangbang: Once all the **checklist** is **done** you have to:
  * **stash merge** this pull request 
  * **delete** the corresponding **branch** 
  * **close** the associated **issue**
## :black_nib: Description

When the Token Map file is not well formatted, instead of crashing, the log now quit gracefully and add a CRITICAL log

Resolve #75 

## :page_with_curl: Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## :heavy_check_mark: Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand 
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

</br>

- [x] :raising_hand: **I certify on my honor that all the informations provided are true, and I've done all I can to deliver a high quality code**

### CODE REVIEW

File: samples/fdga/Generator.cpp [hunk 1]
```diff
@@ -184,5 +209,7 @@ Generator::CreateTask(boost::asio::local::stream_protocol::socket& socket,
 }
 
 Generator::~Generator() {
-    faup_terminate(_faup_handler);
+    if(_faup_handler){
```
Comment (NS4nti) [inline samples/fdga/Generator.cpp]: Careful when merging @HugoSoszynski

File: tests/filters/fconnection.py [hunk 2]
```diff
@@ -56,7 +56,9 @@ def new_connection_test():
     connection_filter.configure()
 
     # START FILTER
-    connection_filter.valgrind_start()
+    if not connection_filter.valgrind_start():
+        connection_filter.clean_files()
+        return False
```
[the remark this answers is not in the document]
Comment (NS4nti) [reply tests/filters/fconnection.py]: done !

File: tests/filters/flogs.py [hunk 3]
```diff
@@ -100,7 +100,9 @@ def run():
 def single_log_to_file():
     filter = Logs()
     filter.configure(FLOGS_CONFIG_FILE)
-    filter.valgrind_start()
+    if not filter.valgrind_start():
+        filter.clean_files()
```
[the remark this answers is not in the document]
Comment (NS4nti) [reply tests/filters/flogs.py]: done !

File: tests/filters/fhostlookup.py [hunk 4]
```diff
@@ -54,7 +54,9 @@ def test(test_name, init_data, data, expected_certitudes, expected_certitudes_si
     hostlookup_filter.configure()
 
     # START FILTER
-    hostlookup_filter.valgrind_start()
+    if not hostlookup_filter.valgrind_start():
+        hostlookup_filter.clean_files()
```
[the remark this answers is not in the document]
Comment (NS4nti) [reply tests/filters/fhostlookup.py]: done !

File: tests/filters/fdga.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    # CONFIG
+    dga_filter = DGA()
+    dga_filter.init_tokens(["a,1",
+                            "b2",
+                            "c,3",
+                            "d,4"])
+    dga_filter.configure()
+
+    # START FILTER
+    dga_filter.valgrind_start()
```
[the remark this answers is not in the document]
Comment (HugoSoszynski) [reply tests/filters/fdga.py]: I did a test with dga and valgrind on a pretty powerful VM.

```
$ time valgrind ./darwin_dga
...
real	0m10,688s
user	0m8,595s
sys	0m0,674s
```

The 3 second sleep after a valgrind start might not be enough.

Moreover the **valgrind check will always fail** due to memory leaks in the Tensorflow version used.
Maybe we can make an **exception** around this filter to enable testing other filters with valgrind without polluting the output with errors **we cannot fix**.

File: tests/filters/fdga.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                            "b,2",
+                            "c,3",
+                            "d,4"])
+    dga_filter.configure()
+
+    # START FILTER
+    if not dga_filter.start():
+        return False
+
+    if not dga_filter.start():
```
Comment (HugoSoszynski) [inline tests/filters/fdga.py]:
```suggestion
    if not dga_filter.stop():
```

**Pull request merged**